osu.Framework/Graphics/Textures/TextureAtlas.cs

```diff
@@ -107,6 +107,9 @@ internal TextureGL Add(int width, int height)
         {
             lock (textureRetrievalLock)
             {
+                if (width > atlasWidth || height > atlasHeight)
+                    throw new InvalidOperationException($"Size of texture to be loaded exceeds the size of the texture atlas: {width}x{height}px.");
```

Comment:
I'd include the incoming size too. The way you phrased it makes it sound like you're outputting the atlas size but it's actually the requested allocation size.

```suggestion
                    throw new InvalidOperationException($"Size of allocation request ({width}x{height}) exceeds the maximum size of this {nameof(TextureAtlas)} ({atlasWidth}x{atlasHeight}).");
```